File: frontend-public/lib/types/workflow.ts
export interface WorkflowNode {
  id?: string
  node_type: string
  label: string
  config?: Record<string, unknown>
  position_x?: number
  position_y?: number
}

export interface WorkflowEdge {
  id?: string
  source_node_id: string
  target_node_id: string
}

export interface Workflow {
  id: string
  owner_id: string
  name: string
  description: string
  status: string
  nodes: WorkflowNode[]
  edges: WorkflowEdge[]
  created_at?: string
  updated_at?: string
}

export interface WorkflowCreatePayload {
  name: string
  description?: string
}

export interface WorkflowUpdatePayload {
  name: string
  description?: string
  nodes: WorkflowNode[]
  edges: WorkflowEdge[]
  status?: string
}

File: src/backend/core/shared/models/workflow.py
"""Workflow 领域模型。"""

from __future__ import annotations

from dataclasses import dataclass, field
from datetime import datetime
from typing import Any, Sequence


@dataclass
class WorkflowNode:
    """工作流节点领域对象。"""

    id: str
    node_type: str
    label: str
    config: dict[str, Any] = field(default_factory=dict)
    position_x: float = 0.0
    position_y: float = 0.0


@dataclass
class WorkflowEdge:
    """工作流边领域对象。"""

    id: str
    source_node_id: str
    target_node_id: str


@dataclass
class Workflow:
    """工作流领域对象。"""

    id: str
    owner_id: str
    name: str
    description: str
    status: str = "draft"
    nodes: Sequence[WorkflowNode] = field(default_factory=list)
    edges: Sequence[WorkflowEdge] = field(default_factory=list)
    created_at: datetime | None = None
    updated_at: datetime | None = None
